lib/vector-core/src/event/vrl_target.rs

```diff
@@ -327,6 +327,137 @@ impl From<Event> for VrlTarget {
     }
 }
 
+#[derive(Debug, Clone)]
+pub enum VrlImmutableTarget<'a> {
+    LogEvent(&'a LogEvent),
+    Metric(&'a Metric),
+    Trace(&'a TraceEvent),
+}
+
+impl<'a> VrlImmutableTarget<'a> {
+    pub fn new(event: &'a Event) -> Self {
+        match event {
+            Event::Log(event) => VrlImmutableTarget::LogEvent(event),
+            Event::Metric(event) => VrlImmutableTarget::Metric(event),
+            Event::Trace(event) => VrlImmutableTarget::Trace(event),
+        }
+    }
+}
+
+impl<'a> vrl_lib::Target for VrlImmutableTarget<'a> {
+    fn target_insert(&mut self, _path: &LookupBuf, _value: vrl_lib::Value) -> Result<(), String> {
+        Err("cannot modify immutable target".to_string())
```

Comment:
If my memory is correct, assignments cannot fail, so I assume this will translate into assigning a null value at runtime?

Reply:
Or I guess the assignment will simply not occur, right?

Reply:
> Or I guess the assignment will simply not occur, right?

See https://github.com/vectordotdev/vector/pull/11916#issuecomment-1076162236